api/v1alpha1/gitlab.go

```diff
@@ -3,9 +3,10 @@ package v1alpha1
 import "k8s.io/apimachinery/pkg/util/intstr"
 
 type GitlabProject struct {
-	// Project specifies the Gitlab group and project (separated by slash) to use
+	// Project specifies the Gitlab group and project (separated by slash) to
+	// use, or the numeric project id
 	// +required
-	Project string `json:"project"`
+	Project *intstr.IntOrString `json:"project"`
```

Comment:
I wonder why this had to be a pointer? It's `required` anyway, so can't be `nil`. It can still be explicitly `""`, which is the thing that needs to be checked.

Reply:
I'd thought that as well, but ultimately chose a pointer to make it consistent with `GitlabMergeRequestRef.MergeRequestId` -- that field is similarly required and a pointer.  No objections here if you'd prefer it not be.

Reply:
Hmm true, looks like I missed that one in the past. Then I'm fine with this being a pointer as well. I'll consider "fixing" each case of this in a future api version bump.